File: PYTHON/Prime-Number-of-Set-Bits-in-Binary-Representation.py
1class Solution(object):
2    def countPrimeSetBits(self, left, right):
3        """
4        :type left: int
5        :type right: int
6        :rtype: int
7        """
8        prime = [2,3,5,7,11,13,17,19,23,29]
9        count=0
10        for i in range(left,right+1):
11            binary = bin(i)[2:]
12            ones = binary.count('1')
13            if ones in prime:
14                count+=1
15        return count
16        

File: JAVA/Prime-Number-of-Set-Bits-in-Binary-Representation.java
1class Solution {
2    public int countPrimeSetBits(int left, int right) {
3        int count =0,ones;
4        for(int i =left; i<=right;i++)
5            if(isPrime(Integer.bitCount(i))) 
6                count++;
7        return count; 
8    }
9    public boolean isPrime(int number){
10        return (number == 2 || number == 3 || number == 5 || number == 7 || number == 11 || number == 13 || number == 17 || number == 19);
11    }
12}